src/main/java/io/vertx/core/logging/LoggerFactory.java

```diff
@@ -46,18 +46,55 @@ public static synchronized void initialise() {
     }
 
     if (className != null) {
-      ClassLoader loader = Thread.currentThread().getContextClassLoader();
+      ClassLoader tcccl = Thread.currentThread().getContextClassLoader();
       try {
-        Class<?> clz = loader.loadClass(className);
-        delegateFactory = (LogDelegateFactory) clz.newInstance();
-      } catch (Exception e) {
-        throw new IllegalArgumentException("Error instantiating transformer class \"" + className + "\"", e);
+        newDelegateFactory = loadClass(className, tcccl);
+      } catch (IllegalArgumentException e) {
+        ClassLoader thisClasssLoader = LoggerFactory.class.getClassLoader();
+        newDelegateFactory = loadClass(className, thisClasssLoader);
       }
     } else {
-      delegateFactory = new JULLogDelegateFactory();
+      newDelegateFactory = new JULLogDelegateFactory();
     }
 
-    LoggerFactory.delegateFactory = delegateFactory;
+    LoggerFactory.delegateFactory = newDelegateFactory;
+  }
+
+  private static LogDelegateFactory loadClass(String className, ClassLoader classLoader) {
+    try {
+      Class<?> clz = classLoader.loadClass(className);
+      return (LogDelegateFactory) clz.newInstance();
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Error instantiating transformer class \"" + className + "\"", e);
```

Comment:
Why _transformer_?

Reply:
dunno, but did you noticed I've just moved this (from line 54), did not actually write it... but yeah, it does looks weird, probably an old copy/paste mistake; so I've just changed it as part of this commit.